### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by daun**
**Description**

- Port to TypeScript
- Switch to cli for bundling
- Update readme

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 <!-- ## [Unreleased] -->
```
Comment (hirasso) [inline CHANGELOG.md]: :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,7 +16,8 @@
 
 - Initial release
 
-[Unreleased]: https://github.com/swup/theme/compare/2.0.0...HEAD
+[Unreleased]: https://github.com/swup/theme/compare/2.1.0...HEAD
```
Comment (hirasso) [inline CHANGELOG.md]: This can also be removed, right?

File: package.json [hunk 3]
```diff
@@ -1,22 +1,28 @@
...
-  "version": "2.0.0",
+  "version": "2.1.0",
   "description": "Base class for creating swup themes",
   "type": "module",
-  "source": "src/index.js",
+  "source": "src/index.ts",
   "main": "./dist/index.cjs",
   "module": "./dist/index.module.js",
-  "exports": "./dist/index.modern.js",
+  "exports": {
```
Comment (hirasso) [inline package.json:12]: Do we need "types" here?

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{
+  "include": ["src"],
+  "compilerOptions": {
+    "target": "ESNext",                                  /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */
+    "module": "Node16",                                  /* Specify what module code is generated. */
```
Comment (hirasso) [inline tsconfig.json]: I started replacing Node16 with NodeNext everywhere. Apparently it's more future proof

**Review by hirasso — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 4 above)
Comment (daun) [reply tsconfig.json]: Is there a tsconfig that I can safely copy and use here? Fragment Plugin?

File: package.json (hunk 3 above)
Comment (daun) [reply package.json:12]: Probably :)

File: tsconfig.json (hunk 4 above)
Comment (hirasso) [reply tsconfig.json]: [Swup's tsconfig](https://github.com/swup/swup/blob/master/tsconfig.json) is the one I would use for new TypeScript projects.

Comment (daun) [reply tsconfig.json]: Copied the fragment plugin tsconfig :)

Comment (hirasso) [reply tsconfig.json]: ok :)

Comment (daun) [reply tsconfig.json]: Just checked and it looks like they're identical.

**Review by hirasso — DISMISSED** [review]
🎓

**Comment by daun** [comment]
Urgh, the new tsconfig broke things.

**Comment by daun** [comment]
Now it works. Again please 👽

**Comment by hirasso** [comment]
Luck! If you activate `declarationMap`, you should also put `/src` in the `files` array in the package.json.

**Comment by daun** [comment]
> Luck! If you activate `declarationMap`, you should also put `/src` in the `files` array in the package.json.

Why? I don't know what that means 🤠

**Comment by daun** [comment]
Added `src` files to npm output.

**Comment by daun** [comment]
> Why? I don't know what that means 🤠

Nevermind, just remembered what this does.

**Review by hirasso — APPROVED** [review]
🎬

**Pull request merged**